packages/node_modules/@webex/plugin-messages/src/messages.js

```diff
@@ -342,74 +339,16 @@ const Messages = WebexPlugin.extend({
    * @private
    * @memberof Messages
    * @param {Object} activity from mercury
-   * @param {Object} event type of "webhook" event
+   * @param {UUID} activity.id of the activity
+   * @param {Object} activity.target the room containing the activity
+   * @param {String} activity.target.url the url location of the room
    * @returns {Object} constructed event
    */
-  getMessageEvent(activity, event) {
-    try {
-      const sdkEvent = cloneDeep(this.eventEnvelope);
-      const cluster = getHydraClusterString(this.webex, activity.target.url);
+  getMessageEvent({id, target: {url}}) {
+    const cluster = getHydraClusterString(this.webex, url);
 
-      sdkEvent.event = event;
-      if (event !== SDK_EVENT.EXTERNAL.EVENT_TYPE.DELETED) {
-        sdkEvent.data.created = activity.published;
-      }
-      sdkEvent.actorId = constructHydraId(hydraTypes.PEOPLE, activity.actor.entryUUID, cluster);
-      sdkEvent.data.roomType =
-        activity.target.tags.includes(SDK_EVENT.INTERNAL.ACTIVITY_TAG.ONE_ON_ONE) ?
-          SDK_EVENT.EXTERNAL.SPACE_TYPE.DIRECT :
-          SDK_EVENT.EXTERNAL.SPACE_TYPE.GROUP;
-      sdkEvent.data.roomId =
-        constructHydraId(hydraTypes.ROOM, activity.target.id, cluster);
-      sdkEvent.data.personId =
-        constructHydraId(hydraTypes.PEOPLE, activity.actor.entryUUID, cluster);
-      sdkEvent.data.personEmail =
-        activity.actor.emailAddress || activity.actor.entryEmail;
-
-      if (event !== SDK_EVENT.EXTERNAL.EVENT_TYPE.DELETED) {
-        const {content, displayName, cards} = activity.object;
-        const text = content || displayName;
-        const files = getHydraFiles(activity);
-
-        sdkEvent.data.id = constructHydraId(hydraTypes.MESSAGE, activity.id, cluster);
-        if (text) {
-          sdkEvent.data.text = text;
-        }
-        if (displayName && displayName !== sdkEvent.data.text) {
-          sdkEvent.data.markdown = displayName;
-        }
-        if (files && files.length) {
-          sdkEvent.data.files = files;
-        }
-        if (cards && cards.length) {
-          sdkEvent.data.attachments = [];
-          for (const card of cards) {
-            sdkEvent.data.attachments.push({
-              contentType: SDK_EVENT.EXTERNAL.ATTACHMENTS.CARD_CONTENT_TYPE,
-              content: JSON.parse(card)
-            });
-          }
-        }
-        if (activity.parent && activity.parent.id) {
-          sdkEvent.data.parentId = constructHydraId(hydraTypes.MESSAGE, activity.parent.id, cluster);
-        }
-      }
-      else {
-        sdkEvent.data.id =
-          constructHydraId(hydraTypes.MESSAGE, activity.object.id, cluster);
-      }
-
-      return sdkEvent;
-    }
-    catch (e) {
-      this.webex.logger.error(`Unable to generate SDK event from mercury \
-'socket activity for message:${event} event: ${e.message}`);
-
-      return null;
-    }
+    return this.get(buildHydraMessageId(id, cluster));
```

Comment:
What a refactor! 😮
Are there any tests to do regression on the features?

Reply:
Integration tests cover most of these. To test the adaptive card:
- add `cardsampler@webex.bot` to a space
- press the `show sample` button.
- check the property `attachments[0].content.body[1].columns[0].items[0].url` is the actual `url` and not a number